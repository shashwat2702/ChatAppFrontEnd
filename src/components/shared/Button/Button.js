import React from 'react';
import './Button.scss';

export default function Button(props) {
  return (
    <div className="buttonComponent">
        <button className="button button-block"
            type={props.type}
            onClick={props.onClick}
        >
            {props.label}
        </button>
    </div>
  )
}
