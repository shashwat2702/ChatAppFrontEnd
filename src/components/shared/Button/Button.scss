$bold: 600;
$main: #1ab188;
$white: #ffffff;
.buttonComponent {
    .button {
        border:0;
        outline:none;
        padding:15px 0;
        font-size:2rem;
        font-weight:$bold;
        text-transform:uppercase;
        letter-spacing:.1em;
        background:$main;
        color:$white;
      }
      
      .button-block {
        display:block;
        width:100%;
      }
}