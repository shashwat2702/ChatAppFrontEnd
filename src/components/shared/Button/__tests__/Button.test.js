import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../Button';

jest.mock('react-router-dom');

describe('Button Component snapshot', () => {
  const props = {
    label: 'Submit',
    type: 'button',
  };
  it('should render correctly', () => {
    const HeaderSnap = renderer.create(<Button {...props} />);
    const HeaderJson = HeaderSnap.toJSON();
    expect(HeaderJson).toMatchSnapshot();
  });
});
describe('Button Component', () => {
  xit('should call onClick when button is clicked', () => {
    const onClickMock = jest.fn();
    const props = {
      onClick: onClickMock,
      label: 'abc',
      type: 'button',
    };
    const wrapper = shallow(<Button {...props} />);
    wrapper.simulate('click');
    expect(onClickMock).toHaveBeenCalled();
  });
});
