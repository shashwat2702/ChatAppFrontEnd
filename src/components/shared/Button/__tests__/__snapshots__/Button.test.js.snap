// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot should render correctly 1`] = `
<div
  className="buttonComponent"
>
  <button
    className="button button-block"
    type="button"
  >
    Submit
  </button>
</div>
`;
